numpy/core/records.py

```diff
@@ -677,7 +678,13 @@ def fromrecords(recList, dtype=None, shape=None, formats=None, names=None,
 
     try:
         retval = sb.array(recList, dtype=descr)
-    except TypeError:  # list of lists instead of list of tuples
+    except (TypeError, ValueError):
+        # list of lists instead of list of tuples ?
+        # 2018-02-07, 1.14.1
+        warnings.warn(
+            "Bad records, may be list of lists instead of list of tuples, "
+            "Trying to fix. In the future this will raise an error",
```

Comment:
I understand but I think it might be a bit unclear. What about "fromrecords expected a list of tuples but got a list of lists. In the future this will raise an error, please update your code to use a list of tuples."

Reply:
Should this warning be emitted if the below code fails?

@ahaldane: _may_ be a list of lists. Who knows why the `TypeError` was actually thrown...

Reply:
@eric-wieser I thought getting both a warning followed by an error wouldn't be too bad if a bit ugly. The alternative is nested try blocks, which I suppose isn't so bad either.

Reply:
ACtually, warning followed by error is better - if the user has warnings as errors, then they'll see the more useful warning, rather than a more cryptic error.